services/core/SQLHistorian/tests/test_sqlitehistorian_unit.py

```diff
@@ -0,0 +1,92 @@
+import os
+from shutil import rmtree
+import subprocess
+
+import pytest
+from gevent import sleep
+from datetime import timedelta
+from services.core.SQLHistorian.sqlhistorian import historian
+
+CACHE_NAME = "backup.sqlite"
+HISTORIAN_DB = "./data/historian.sqlite"
+
+
```

Comment:
There is a separate  test file for base historian under volttrontesting/services/historian/test_bbase_historian.py . This test case should get added to base historian tests

Reply:
note this is a unit test not an integration test.  it does not run the platform.

Reply:
To @craig8's point, the intent of this test was to verify that a concrete a historian (in this case, an historian using sqlite) was handling duplicates as expected, i.e. that the last duplicate "wins" and gets saved in the db.

Reply:
This a good one to test. But since this feature is added to base historian it would be good to add a test case to base historian test cases irrespective of the concrete historian.

Reply:
Agree. I added this extra test to validate our assumption that the "last one" wins in a duplicate saved to a database at the concrete historian level. The initial scope for this PR is fixing the backup_database so that duplicates aren't sent to the base_historian. I'll create a separate issue for base_historian unit tests to be tracked in a different PR.

Reply:
Added a simple base historian agent unit test that mocks the parent agent in the latest commit, 46321f8.